src/main/java/com/ericsson/ei/waitlist/WaitListWorker.java

```diff
@@ -65,23 +67,35 @@ public void run() {
         if(shutdownInProgress) {
             return;
         }
+        try {
+            getAllDocumentsAndCheckTargetAggregations();
+        } catch (MongoClientException e) {
+            LOGGER.error("Failed to get documents from MongoDB", e.getMessage());
+        }
+    }
+
+    private void getAllDocumentsAndCheckTargetAggregations() {
         List<String> documents = waitListStorageHandler.getWaitList();
         for (String document : documents) {
             try {
-                ObjectMapper objectMapper = new ObjectMapper();
-                JsonNode eventJson = objectMapper.readTree(document);
-                String id = eventJson.get("_id").asText();
-                if (eventToObjectMapHandler.isEventInEventObjectMap(id)) {
-                    waitListStorageHandler.dropDocumentFromWaitList(document);
-                } else {
-                    checkTargetAggregationsExistAndRepublishEvent(eventJson);
-                }
+                checkAggregationsExistAndRepublishEvent(document);
             } catch (Exception e) {
                 LOGGER.error("Exception occured while trying to resend event: {}", document, e);
             }
         }
     }
 
+    private void checkAggregationsExistAndRepublishEvent(String document) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode eventJson = objectMapper.readTree(document);
+        String id = eventJson.get("_id").asText();
```

Comment:
If it's not too much hassle I would like the key "_id" to be a constant at the top of the class instead. The same goes for the key "Event" used further down.